### PR TITLE
[Chore] Add more files to be deleted to doc make clean target

### LINKED ISSUE #46206: [Doc] Use make clean instead of git clean
### Description

The [troubleshooting section of the ducumentation](https://docs.ray.io/en/latest/ray-contribute/docs.html#troubleshooting) tells us to use `git clean` to clean build artifacts instead of `make clean`. However, this only works for `conda` environment. If developers use `pyenv` or `virtualenv`, the  `.venv/` folder and the `.python-version` file will be deleted, which causes problem. Therefore, it is better to write down which files to be deleted more completely in `Makefile` `clean` target instead of asking developers to run `git clean`.

Note: I am willing to work on this issue.

### Link

_No response_

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Why are these changes needed?

See the description of the corresponding issue for details.

## Related issue number

Closes #46206 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by peytondmurray** [comment]
Thanks, this is convenient to have :rocket:

**Review by peytondmurray — APPROVED** [review]

File: doc/Makefile [hunk 1]
```diff
@@ -55,12 +55,19 @@ help:
...
+	rm -rf ./source/_ext/__pycache__/
 	rm -rf ./source/*/api/doc/*
 	rm -rf ./source/ray-references/api/*/doc/*
 	rm -rf ./source/cluster/running_applications/doc/*
 	rm -rf ./source/cluster/running_applications/job-submission/doc/*
 	rm -rf ./source/ray-observability/api/state/doc*
 	rm -rf ./source/rllib/package_ref/doc*
+	rm -rf ./source/ray-more-libs/doc/*
+	rm -rf ./source/ray-observability/reference/doc/*
+	rm -f ./source/data/examples.rst
```
Comment (can-anyscale) [inline doc/Makefile:68]: a bit weird that something is generating files directly on a working directory, but unrelated to your code

**Review by can-anyscale — APPROVED** [review]

File: doc/Makefile (hunk 1 above)
Comment (peytondmurray) [reply doc/Makefile:68]: This comes from various `autosummary` directives. Not ideal, but not related to this PR.

**Pull request merged**